### PR TITLE
kv: add StorageEngine config

### PR DESCRIPTION
**Pull request by icorderi**
## Summary

We are  breaking down #5212 in several smaller PRs to make it more digestible. 

This first PR add a storage engine config.
The config version is bumped to v28. 

This is built on top of the following PRs:
- #5216

## Test Plan

existing tests.

### CODE REVIEW

File: ledger/acctdeltas_test.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func checkAccounts(t *testing.T, tx trackerdb.TransactionScope, rnd basics.Round
 	require.NoError(t, err)
 	require.Equal(t, r, rnd)
 
-	aor, err := tx.Testing().MakeAccountsOptimizedReader()
+	aor, err := tx.MakeAccountsOptimizedReader()
```
Comment (algonautshant) [inline ledger/acctdeltas_test.go:62]: Why is this removed here and not elsewhere?
Maybe remove the Testing() from the interface as well? It is not a good idea to have something for testing in the production code.

File: ledger/store/trackerdb/store.go [hunk 2]
```diff
@@ -48,6 +48,7 @@ type TransactionScope interface {
 	MakeCatchpointReaderWriter() (CatchpointReaderWriter, error)
 	MakeAccountsReaderWriter() (AccountsReaderWriter, error)
 	MakeAccountsOptimizedWriter(hasAccounts, hasResources, hasKvPairs, hasCreatables bool) (AccountsWriter, error)
+	MakeAccountsOptimizedReader() (AccountsReader, error)
```
[the remark this answers is not in the document]
Comment (icorderi) [reply ledger/store/trackerdb/store.go:51]: The SQL code has 2 types of clients, one with prepare statements in it.
That one is called "Optimized". 
This comes from the early work refactoring the SQL code into interfaces.

File: ledger/acctdeltas.go [hunk 3]
```diff
@@ -476,29 +476,13 @@ func (a *compactAccountDeltas) accountsLoadOld(tx trackerdb.TransactionScope) (e
...
-			// Note: the err will be ignored in this case since `err` is being shadowed.
-			// this behaviour is equivalent to `err = nil`
-		default:
+		data, err := arw.LookupAccount(addr)
+		if err != nil {
 			// unexpected error - let the caller know that we couldn't complete the operation.
 			return err
 		}
+		// update the account
+		a.updateOld(idx, data)
```
[the remark this answers is not in the document]
Comment (icorderi) [reply ledger/acctdeltas.go:485]: This is part of https://github.com/algorand/go-algorand/pull/5216
I'm cutting the KV PRs on top of each other.

File: ledger/acctdeltas_test.go (hunk 1 above)
Comment (icorderi) [reply ledger/acctdeltas_test.go:62]: This comes from early refactoring too. 
The methods used only in testing are behind a testing interface. 
`MakeAccountsOptimizedReader` was promoted and removed from the testing only.

**Review by algorandskiy — APPROVED** [review]
